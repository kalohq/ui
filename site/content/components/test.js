<h1 id="design-tokens">Design Tokens</h1>
<blockquote>
<p><strong>Note</strong>: The files in this directory are autogenerated. See <a href="/config/design-tokens">config/design-tokens</a> for the source.</p>
</blockquote>
