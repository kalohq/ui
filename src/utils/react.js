import ReactDOM from 'react-dom';

/**
 * Focus a ref’d react input when it is mounted
 * @param ref
 */
export function focusOnMount(ref) {
  const el = ReactDOM.findDOMNode(ref);

  if (el) {
    // Give focus and put the cursor at the end of
    // the value.
    el.focus();
    el.value = el.value;
  }
}

/**
 * Select the text of a ref’d react input when it is mounted
 * @param ref
 */
export function selectOnMount(ref) {
  if (ref && ref.select) {
    ref.select();
  }
}
