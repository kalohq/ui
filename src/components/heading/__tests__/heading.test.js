/* eslint-env jest */
import * as React from 'react';
import renderer from 'react-test-renderer';

import {shallow, mount} from 'utils/test/enzyme';

import Heading from '../heading.react';

describe('Heading', () => {
  const create = props => <Heading {...props}>A Heading Component</Heading>;

  it('should match snapshot', () => {
    const element = renderer
      .create(<Heading size="medium">A medium sized header</Heading>)
      .toJSON();
    expect(element).toMatchSnapshot();
  });

  it('should allow alignment to be set', () => {
    const element = create({
      align: 'right',
    });

    const result = shallow(element);
    expect(result.hasClass('heading--align-right')).toBe(true);
  });

  it('should allow a color from the palette to be set', () => {
    const element = create({
      color: 'blue500',
    });

    const result = shallow(element);
    expect(result.hasClass('color-blue500')).toBe(true);
  });

  it('should allow the DOM element to be changed', () => {
    const element = create({
      component: 'H1',
    });

    const result = mount(element);
    expect(result.html()).toContain('<h1');
  });
});
