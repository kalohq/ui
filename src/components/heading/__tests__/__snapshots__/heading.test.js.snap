// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should match snapshot 1`] = `
<span
  className="heading heading--medium heading--align-unset heading--ellipsis color-navy700 e1q9f2xp6 css-l4muhg"
  style={Object {}}
>
  A medium sized header
</span>
`;
